sys/kern/sysv_sem.c

```diff
@@ -2210,7 +2125,7 @@ freebsd7_freebsd64___semctl(struct thread *td,
 		break;
 	case GETALL:
 	case SETALL:
-		semun.array = __USER_CAP_ARRAY(arg.array, SHRT_MAX);
+		semun.array = __USER_CAP_UNBOUND(arg.array);
```

Comment:
```suggestion
		semun.array = __USER_CAP(arg.array,
		    SHRT_MAX * sizeof(*semun.array));
```

Reply:
I almost did that, but the actual freebsd64__semctl version below uses __USER_CAP_UNBOUND (and that's the version that is actually used by contemporary 64-bit binaries)

Reply:
Should probably improve that then?

Reply:
So I'm testing a commit to change both of these limits to use seminfo.semmns (boot time tunable, so read-only post boot) instead of SHRT_MAX as a better upper bound on the array size.